### PR TITLE
Add Microsoft.DotNet.HotReload.Utils.Generator.BuildTool MSBuild target nuget

### PR DESCRIPTION
**Pull request by lambdageek**
Make a nuget package that has a target that runs a copy of the delta generator right after a `Build` target.

Factor out the common command-line frontend into a new Frontend assembly.

### CODE REVIEW

**Comment by lambdageek** [comment]
FYI: @ViktorHofer you were right, adding a targets nuget wasn't too bad.  Tried it locally, seems to be working.  After I merge this and publish packages, I'll update the dotnet/runtime#51144 to use it

File: src/Microsoft.DotNet.HotReload.Utils.Generator.BuildTool/Microsoft.DotNet.HotReload.Utils.Generator.BuildTool.csproj [hunk 1]
```diff
@@ -0,0 +1,19 @@
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <Nullable>enable</Nullable>
+    <TargetFramework>net6.0</TargetFramework>
+    <!-- FIXME: do I need ExcludeFromSourceBuild? -->
+    <ExcludeFromSourceBuild>true</ExcludeFromSourceBuild>
```
Comment (ViktorHofer) [inline src/Microsoft.DotNet.HotReload.Utils.Generator.BuildTool/Microsoft.DotNet.HotReload.Utils.Generator.BuildTool.csproj:7]: I don't think package needs to be built in sourcebuild as it's only used for building tests, right? So you can probably remove the property.

File: src/Microsoft.DotNet.HotReload.Utils.Generator.BuildTool/Microsoft.DotNet.HotReload.Utils.Generator.BuildTool.csproj [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    <PackageType>MSBuildSdk</PackageType>
+    <IncludeBuildOutput>fasle</IncludeBuildOutput>
+    <IsPackable>true</IsPackable>
```
Comment (ViktorHofer) [inline src/Microsoft.DotNet.HotReload.Utils.Generator.BuildTool/Microsoft.DotNet.HotReload.Utils.Generator.BuildTool.csproj:10]: You don't need any of these properties as those are already set in BuildTask.targets.

```suggestion
```

File: src/Microsoft.DotNet.HotReload.Utils.Generator.BuildTool/build/Microsoft.DotNet.HotReload.Utils.Generator.BuildTool.targets [hunk 3]
```diff
@@ -0,0 +1,37 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (ViktorHofer) [inline src/Microsoft.DotNet.HotReload.Utils.Generator.BuildTool/build/Microsoft.DotNet.HotReload.Utils.Generator.BuildTool.targets:1]:
```suggestion
```

The xml isn't needed anymore.

File: src/Microsoft.DotNet.HotReload.Utils.Generator.BuildTool/build/Microsoft.DotNet.HotReload.Utils.Generator.BuildTool.targets [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+<!-- Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. -->
+<Project DefaultTargets="Build">
```
Comment (ViktorHofer) [inline src/Microsoft.DotNet.HotReload.Utils.Generator.BuildTool/build/Microsoft.DotNet.HotReload.Utils.Generator.BuildTool.targets:3]:
```suggestion
<Project>
```

a DefaultTarget isn't needed.

File: src/Microsoft.DotNet.HotReload.Utils.Generator.Frontend/Microsoft.DotNet.HotReload.Utils.Generator.Frontend.csproj [hunk 5]
```diff
@@ -0,0 +1,12 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <OutputType>Library</OutputType>
```
Comment (ViktorHofer) [inline src/Microsoft.DotNet.HotReload.Utils.Generator.Frontend/Microsoft.DotNet.HotReload.Utils.Generator.Frontend.csproj:3]:
```suggestion
```

Library is the default OutputType.

File: src/Microsoft.DotNet.HotReload.Utils.Generator.Frontend/Frontend.cs [hunk 6]
```diff
@@ -0,0 +1,114 @@
+using System;
```
Comment (ViktorHofer) [inline src/Microsoft.DotNet.HotReload.Utils.Generator.Frontend/Frontend.cs:1]: Missing copyright header

File: eng/BuildTask.targets [hunk 7]
```diff
@@ -0,0 +1,84 @@
+<Project>
```
Comment (ViktorHofer) [inline eng/BuildTask.targets:1]: I would leave a comment here to indicate that this code is copied from Arcade.

File: src/hotreload-delta-gen/src/hotreload-delta-gen/Program.cs [hunk 8]
```diff
@@ -1,116 +1,2 @@
...
-            return true;
-        }
-
-        private static bool Xor (bool a, bool b) {
-            return !(a == b);
-        }
-
-    }
-}
+﻿
```
Comment (ViktorHofer) [inline src/hotreload-delta-gen/src/hotreload-delta-gen/Program.cs:1]: Missing copyright header.

File: src/Microsoft.DotNet.HotReload.Utils.Generator.BuildTool/Program.cs [hunk 9]
```diff
@@ -0,0 +1,2 @@
+﻿
```
Comment (ViktorHofer) [inline src/Microsoft.DotNet.HotReload.Utils.Generator.BuildTool/Program.cs:1]: Missing copyright header.

**Pull request merged**